library/core/src/init.rs

```diff
@@ -0,0 +1,52 @@
+//! In-place initialization
+//!
+//! This module describes the interface through which types supporting in-place initialization
+//! interact with allocation mechanism to ensure correct and safe initialization of values
+//! within the memory slots provided by the allocation.
```

Comment:
I don't have a good suggestion here, but this reads a bit weird.

Reply:
Hmm let me try to rephrase it. Or let me leave a generic note, given that there may be more to come into this module.